src/bun.js/api/server.zig

```diff
@@ -1400,6 +1403,12 @@ fn NewRequestContext(comptime ssl_enabled: bool, comptime debug_mode: bool, comp
                 this.allocator.free(bun.constStrToU8(this.pathname));
                 this.pathname = "";
             }
+
+            // if we are waiting for the body yet and the request was not aborted we can safely clear the onData callback
+            if (this.is_waiting_body and this.aborted == false) {
+                this.resp.clearOnData();
```

Comment:
Are you sure that `resp` is still valid memory here? We don't access `resp` in the other functions here. `endWithoutBody` makes the `resp` invalid memory (I think?) and that is called when using sendfile() before this function is called 

https://github.com/oven-sh/bun/blob/945d069290fcf0ce852120b809ed8aa6cdd1e245/src/bun.js/api/server.zig#L1454

Reply:
I will refactor this https://github.com/oven-sh/bun/pull/2803/commits/92e9530f0a2e5cc39e6e3864edc3fa32049fbd7e#diff-e9484fda309708f80db81a125281dd0b9fdadaf3d2400ae8e78a2b8fc14d85eb and make resp optional to avoid the footgun